pkgs/top-level/python-packages.nix

```diff
@@ -7527,7 +7527,7 @@ self: super: with self; {
     abseil-cpp = if stdenv.isLinux then
         pkgs.abseil-cpp_202301
       else
-        pkgs.abseil-cpp_202301.override { cxxStandard = "17"; };
+        pkgs.abseil-cpp_202301.override { cxxStandard = "14"; };
```

Comment:
While we're here, the comment above "on darwin clang is on an old version so c++ defaults to 11" should say clang defaults to 14, right? https://github.com/NixOS/nixpkgs/pull/255844#issuecomment-1724533763

Reply:
I am going to comment in the new PR, but in the original error I posted, I get warnings that default c++ is 11 on darwin actually.
https://github.com/onnx/onnx/issues/5579#issuecomment-1716179816
(after the new PR made by tobi, the default will be 14 though but I think we can keep the comment in the meanwhile.